host-bmc/dbus/pcie_topology.cpp

```diff
@@ -0,0 +1,130 @@
+#include "pcie_topology.hpp"
+
+#include "libpldm/entity.h"
+#include "oem/ibm/libpldm/state_set_oem_ibm.h"
+
+#include "host-bmc/dbus/custom_dbus.hpp"
+
+namespace pldm
+{
+namespace dbus
+{
+
+bool PCIETopology::pcIeTopologyRefresh() const
+{
+    return sdbusplus::com::ibm::PLDM::server::PCIeTopology::
+        pcIeTopologyRefresh();
+}
+
+bool PCIETopology::updateTopologyRefresh(bool value)
+{
+    return sdbusplus::com::ibm::PLDM::server::PCIeTopology::pcIeTopologyRefresh(
+        value);
+}
+
+bool PCIETopology::pcIeTopologyRefresh(bool value)
+{
+    std::vector<set_effecter_state_field> stateField;
+
+    if (value ==
+        sdbusplus::com::ibm::PLDM::server::PCIeTopology::pcIeTopologyRefresh())
+    {
+        stateField.push_back({PLDM_NO_CHANGE, 0});
+        return sdbusplus::com::ibm::PLDM::server::PCIeTopology::
+            pcIeTopologyRefresh(value);
+    }
+    else
+    {
+        stateField.push_back({PLDM_REQUEST_SET, GET_PCIE_TOPOLOGY});
+        stateField.push_back({PLDM_REQUEST_SET, GET_CABLE_INFO});
+    }
+
+    if (value && hostEffecterParser)
+    {
+        uint16_t effecterID = getEffecterID();
+
+        if (effecterID == 0)
+        {
+            return false;
+        }
+        hostEffecterParser->sendSetStateEffecterStates(
+            mctpEid, effecterID, 1, stateField,
+            std::bind(
+                std::mem_fn(&pldm::dbus::PCIETopology::updateTopologyRefresh),
+                this, std::placeholders::_1),
+            value);
```

Comment:
ah , pavitra  i missed this  - you need to set the effecters with both 1 and 2

Reply:
done